### DISCUSSION TITLE
How can I know what job has been exceeded memory limit?

### DISCUSSION THREAD

**Discussion by paintmeyellow** [laravel/framework#35427] [Q&A]
After exceeded memory limit job is just stopping??
Is there any way to log it?
`php -d memory_limit=50M artisan queue:work`

**Comment by joelbutcher** [comment]
@Nikitos096 Have you checked out [Laravel Horizon](https://laravel.com/docs/8.x/horizon)?

**Reply by paintmeyellow** [reply]
Yep, it's just stopped and freeze like forever in Pending Jobs

**Reply by joelbutcher** [reply]
Hmm 🤔 

Do you have a stack trace?

**Reply by paintmeyellow** [reply]
Here what I got
- supervisord.conf
![image](https://user-images.githubusercontent.com/24860764/100769147-782f6500-341d-11eb-8ffb-81b316b1d99b.png)

- config/horizon.php
![image](https://user-images.githubusercontent.com/24860764/100769293-a6ad4000-341d-11eb-8deb-401ea9b4b5f1.png) ![image](https://user-images.githubusercontent.com/24860764/100769342-b6c51f80-341d-11eb-8b20-ea9c5e4de42d.png)

- config/queue.php
![image](https://user-images.githubusercontent.com/24860764/100769872-70bc8b80-341e-11eb-9a01-1daf28fbe7c5.png) ![image](https://user-images.githubusercontent.com/24860764/100769848-69957d80-341e-11eb-94a6-1133d6cebbae.png)

- Job
![image](https://user-images.githubusercontent.com/24860764/100769569-0dcaf480-341e-11eb-854c-0ac97a8e73b5.png)

The thing is when memory is reaches 1.5G, worker is just stop process.
I can see it in here (htop). One of them is restarting
![image](https://user-images.githubusercontent.com/24860764/100771129-dc532880-341f-11eb-8772-c27e5d1baac0.png)

And I cannot figure out why it's just stop in 1.5G, if memory=256M (config/horizon.php)

Here is the last what I can see in my logs
![image](https://user-images.githubusercontent.com/24860764/100771989-e164a780-3420-11eb-82ca-c6a4f6a01b02.png)

**Comment by paintmeyellow** [comment]
Anyone?

**Comment by Himakar-PV** [comment]
I am also running into similar issue, I want to know when memory leaked and rollback the logic inside job. Currently I can just stop committing but no rollback .
